src/auth/models/strategyFactory.class.ts

```diff
@@ -0,0 +1,29 @@
+import oAuth2 from '../oauth2/models/oauth2.class'
+import openid from '../oidc/models/openid.class'
+import redisSession from '../session/models/redisSessionStrategy.class'
+import fileSession from '../session/models/fileSessionStrategy.class'
+import { Strategy } from './strategy.class'
+import { OAUTH2 } from '../oauth2'
+import { OIDC } from '../oidc'
+import { SESSION } from '../session/session.constants'
+
+export type StrategyType = 'oauth2' | 'oidc' | 's2s' | 'filesession' | 'redissession'
```

Comment:
again a 'session' isn't a strategy/authentication type, can we split this into it's own layer please